### PR TITLE
add user and userAPI skeelton code

### PR DESCRIPTION
**Pull request by dlindsey-olioapps**
**Ticket**
none

**What does this PR do**
This will add the skeleton code for User and UserAPI classes. The User class will be able to eventually be added to an array in UserAPI class.

**Media**
Screen shot

**How to test**
Clone the repo to your computer and navigate to it. Install necessary packages with Yarn, and then launch the application with TypeScript complier via the command "yarn run dev". You should get a result similar to the image above

### CODE REVIEW

**Comment by dlindsey-olioapps** [comment]
**Ticket**
none

**What does this PR do**
This will add the skeleton code for User and UserAPI classes. The User class will be able to eventually be added to an array in UserAPI class.

**Media**
image

**How to test**
Clone the repo to your computer and navigate to it. Install necessary packages with Yarn, and then launch the application with TypeScript complier via the command "yarn run dev". You should get a result similar to the image above

File: src/index.ts [hunk 1]
```diff
@@ -1 +1,49 @@
-//Define class here
\ No newline at end of file
+//Define class here
+class User {
+  name: string;
```
Comment (rachoac) [inline src/index.ts]: no need for semi colons (es6 / typescript conventions)

File: src/index.ts [hunk 2]
```diff
@@ -1 +1,49 @@
...
+  favoriteColor: string;
+  age: number;
+  id: number;
+  constructor(name: string, favoriteColor: string, age: number, id: number) {
```
Comment (rachoac) [inline src/index.ts:7]: add a carriage return before this constructor

**Comment by rachoac** [comment]
> 

you should be adding these details to the body of the PR, not in a comment

**Comment by rachoac** [comment]
> **Ticket** none
> 
> **What does this PR do** This will add the skeleton code for User and UserAPI classes. The User class will be able to eventually be added to an array in UserAPI class.
> 
> **Media** image
> 
> **How to test** Clone the repo to your computer and navigate to it. Install necessary packages with Yarn, and then launch the application with TypeScript complier via the command "yarn run dev". You should get a result similar to the image above

you should add these details to the body of the PR, not in a comment

File: src/index.ts [hunk 3]
```diff
@@ -1 +1,49 @@
...
+}
+
+class UserAPI {
+  users: Array<User>;
+
+  constructor() {
+    this.users = [];
+  }
+
+  assignId() {
```
Comment (DuHerb) [inline src/index.ts]: could make this a `private` method, as it should/will not be exposed

Comment (dlindsey-olioapps) [reply src/index.ts]: Thanks for the tip! Definitely making it private

File: src/index.ts [hunk 4]
```diff
@@ -1 +1,45 @@
...
+  name: string
+  favoriteColor: string
+  age: number
+  id: number
+
+  constructor(name: string, favoriteColor: string, age: number, id: number) {
+    this.name = name;
+    this.favoriteColor = favoriteColor;
+    this.age = age;
+    this.id = id;
```
Comment (rachoac) [inline src/index.ts]: remove all semi colons from this file

**Pull request merged**